src/shared/tpm2-util.c

```diff
@@ -138,13 +136,19 @@ ESYS_TR tpm2_flush_context_verbose(ESYS_CONTEXT *c, ESYS_TR handle) {
         return ESYS_TR_NONE;
 }
 
-int tpm2_context_init(const char *device, Tpm2Context *ret) {
-        _cleanup_(Esys_Finalize_wrapper) ESYS_CONTEXT *c = NULL;
-        _cleanup_free_ TSS2_TCTI_CONTEXT *tcti = NULL;
-        _cleanup_(dlclosep) void *dl = NULL;
+int tpm2_context_new(const char *device, Tpm2Context **ret_context) {
+        _cleanup_tpm2_context_ Tpm2Context *context = NULL;
         TSS2_RC rc;
         int r;
 
+        assert(ret_context);
+
+        context = new0(Tpm2Context, 1);
+        if (!context)
+                return log_oom();
+
+        context->n_ref = 1;
```

Comment:
hmm, instead of using new() and then initializing fields, we prefer nowadays in new code to just use new() but then to use compound initialization, i.e.:

```c
context = new(Tpm2Context, 1);
if (!context)
        return log_oom();

*context = (Tpm2Context) {
        .n_ref = 1,
};
```

this is not just easier on the eye, but should ideally allow the compiler to make better optimizations.